awx/ui_next/src/screens/Credential/shared/CredentialPlugins/CredentialPluginPrompt/MetadataStep.jsx

```diff
@@ -27,6 +28,21 @@ function MetadataStep({ i18n }) {
   const [selectedCredential] = useField('credential');
   const [inputValues] = useField('inputs');
 
+  const {
+    result: testPluginSuccess,
+    error: testPluginError,
+    request: testPluginMetadata,
+  } = useRequest(
+    useCallback(
+      async (credential, metadata) =>
+        CredentialsAPI.test(credential.id, {
+          metadata,
+        }),
```

Comment:
This function should access `selectedCredential.value` and `inputValues.value` directly here, rather than passing them in as parameters (this will prevent re-defening the function on every render down on line 161— use `onClick={testPluginMetadata}` instead)

Reply:
I think this should fix your problem with the error lingering from earlier requests, too

Reply:
This should be address with https://github.com/ansible/awx/pull/7338/commits/0c9df15fa53ca1fb70ca09a34f5029c56d5fd344 iirc